layouts/PortfolioLayout.tsx

```diff
@@ -45,11 +48,21 @@ const PortfolioLayout: React.FC<PropsWithChildren> = ({ children }) => {
           shallow: true,
         },
       );
+      setIsLoading(false);
     } else {
       setHasAddress(false);
+      setIsLoading(false);
     }
   }, [addressFromRoute, router.isReady, wallet.activeAccount?.address]);
 
+  if (isLoading) {
+    return (
+      <div className="flex justify-center items-center w-full min-h-[50vh]">
+        <Loader />
```

Comment:
I was hoping to do that but there is some business logic that occurs on `PortfolioLayout.tsx` with getting the user address and dynamic routing that happens before the portfolio page is loaded; it's during that phase where we see the delay in the portfolio page being initially loaded. Also, in case there is no wallet connected it would not make sense to show the skeletons above either.